### PR TITLE
session: fix tables_priv table schema (#33599)

### PR DESCRIPTION
**Pull request by ti-srebot**
cherry-pick #33599 to release-5.3
You can switch your code base to this Pull Request by using [git-extras](https://github.com/tj/git-extras):
```bash
# In tidb repo:
git pr https://github.com/pingcap/tidb/pull/33604
```

After apply modifications, you can push your change to this PR via:
```bash
git push git@github.com:ti-srebot/tidb.git pr/33604:release-5.3-e15e6d035872
```

---

### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/33588

Problem Summary:

https://github.com/pingcap/tidb/pull/28546 modify `tables_priv` and `columns_priv` tables' column(`Column_priv`) type. But it doesn't update `tables_priv` column in upgrade step.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
deploy a v5.2 cluster
upgrade to this PR.
execute SQL
```
use test;
create table t(a int);
create user u1;
grant all on test.t to u1;
```
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
 Fix the issue that privilege-related operations may fail for upgraded clusters.
```

### CODE REVIEW

**Comment by zimulala** [comment]
PTAL @Defined2014 @bb7133

File: session/bootstrap.go [hunk 1]
```diff
@@ -357,6 +357,14 @@ const (
 		last_analyzed_at TIMESTAMP,
 		PRIMARY KEY (table_id, column_id) CLUSTERED
 	);`
+	// CreateTableCacheMetaTable stores the cached table meta lock information.
+	CreateTableCacheMetaTable = `CREATE TABLE IF NOT EXISTS mysql.table_cache_meta (
```
Comment (bb7133) [inline session/bootstrap.go:361]: Could we remove `CreateTableCacheMetaTable`?

**Review by Defined2014 — APPROVED** [review]

File: session/bootstrap.go (hunk 1 above)
Comment (zimulala) [reply session/bootstrap.go:361]: This may affect the next cherry-pick classmates.

**Review by bb7133 — APPROVED** [review]
LGTM

**Pull request merged**